### PR TITLE
Big piecewise processing chunk of work

### PR DESCRIPTION
**Pull request by jiwalker-usgs**
This is working for the most part, though there is an "off-by-one" sort of error where the edges of the execution plan area is missing a transect.

### CODE REVIEW

File: dsas-web-ui/src/main/webapp/WEB-INF/jsp/stages/transect_verification.jsp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<!-- Transects --> 
+<div class="tab-pane container-fluid" id="transect_verification">
+	<div class="row-fluid">
+		<div class="span4"><h3>Execution Plan</h3></div>
+		<div class="span8" id="transect_verification-alert-container"></div>
+	</div>
+	<div class="tab-pane active" id="transect_verification-view-tab">
+		Calculation Projection: 
+		<select id="ctrl-transect-verification-utm">
+			<option value="EPSG:26910">UTM 10</option>
```
Comment (isuftin) [inline dsas-web-ui/src/main/webapp/WEB-INF/jsp/stages/transect_verification.jsp]: So are these correct, then?

File: dsas-web-ui/src/main/webapp/js/stages/transect_verification.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+		'view-tab': 'Transect Verification View Tab Text.'
+	},
+	appInit: function () {
+		$.get('templates/wps-request-create-execution-plan.mustache').done(function (data) {
+			TransectVerification.WPS_REQUEST_TEMPLATE = Handlebars.compile(data);
+		});
+
+		$(this.controlIdentifiers.submitButton).on('click', $.proxy(this.processCalculation, this));
+
+		this.UTM_LAYER = new OpenLayers.Layer.ArcGIS93Rest("UTM Zones",
```
Comment (isuftin) [inline dsas-web-ui/src/main/webapp/js/stages/transect_verification.js]: Do we want to remove this if it's not currently working?

File: dsas-web-ui/src/main/webapp/js/stages/transect_verification.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+			request: TransectVerification.WPS_REQUEST,
+			context: args.context || this
+		});
+	},
+	processCalculation: function () {
+		$.when(this.submitWPSRequest({
+			wpsRequestObject: this.WPS_REQUEST,
+			context: this
+		})).done(function (data, textStatus, jqXHR) {
+			if (typeof data === 'string') {
```
Comment (isuftin) [inline dsas-web-ui/src/main/webapp/js/stages/transect_verification.js]: We should also check for error here

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/util/CRSUtils.java [hunk 4]
```diff
@@ -292,4 +294,13 @@ public static boolean isNewLineSegment(SimpleFeature first, SimpleFeature second
 		}
 		return isNewSegment;
 	}
+	
+	public static CoordinateReferenceSystem lookupCRS(String epsgCode) throws FactoryException {
+		String lookup = epsgCode;
+		if (!epsgCode.startsWith("EPSG") && !epsgCode.startsWith("epsg")) {
```
Comment (isuftin) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/util/CRSUtils.java]: if (!epsgCode.toLowerCase().startsWith("epsg")) {}

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/util/CRSUtils.java [hunk 5]
```diff
@@ -292,4 +294,13 @@ public static boolean isNewLineSegment(SimpleFeature first, SimpleFeature second
...
+			lookup = "EPSG:" + epsgCode;
+		}
+		CoordinateReferenceSystem crs = CRS.decode(lookup);
```
Comment (isuftin) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/util/CRSUtils.java]: Do you not have a need for [testing with longitude first](http://docs.geotools.org/stable/javadocs/org/geotools/referencing/CRS.html#decode(java.lang.String, boolean)) ?

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/CreateExecutionPlanProcess.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+			}
+		}
+		
+		public String execute() throws Exception {
+			importer.checkIfLayerExists(workspace, planLayer);
+			CoordinateReferenceSystem shorelinesCrs = CRSUtils.getCRSFromFeatureCollection(shorelineFeatureCollection);
+			CoordinateReferenceSystem baselineCrs = CRSUtils.getCRSFromFeatureCollection(baselineFeatureCollection);
+			
+			if (!CRS.equalsIgnoreMetadata(shorelinesCrs, REQUIRED_CRS_WGS84)) {
+				throw new UnsupportedCoordinateReferenceSystemException("Shorelines are not in accepted projection");
```
Comment (isuftin) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/CreateExecutionPlanProcess.java]: Might want to also include the accepted projection in the error response

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/UpdateTransectsAndIntersectionsProcess.java [hunk 7]
```diff
@@ -173,13 +178,14 @@ private String execute() throws Exception {
...
-						updatedTransects.add(updatedTransect);
+						// get the baseline from plan and reproject
+						//Transect updatedTransect = updateTransectBaseDist(transectObj, transformedBaseline);
+						//transectStore.modifyFeatures(new NameImpl(BASELINE_DIST_ATTR), updatedTransect.getBaselineDistance(), transectFilter);
+						//updatedTransects.add(updatedTransect);
 					}
 					transaction.commit();
 				}
-				calc.calculateIntersections(updatedTransects.toArray(new Transect[updatedTransects.size()]), shorelines);
+				//calc.calculateIntersections(updatedTransects.toArray(new Transect[updatedTransects.size()]));
```
Comment (isuftin) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/UpdateTransectsAndIntersectionsProcess.java]: Remove commented code

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java [hunk 8]
```diff
@@ -109,70 +141,77 @@ public IntersectionCalculator(SimpleFeatureCollection shorelines, SimpleFeatureC
...
-					startDistance += subTransectLength;
-				} else {
-					subTransect = transect;
-					startDistance = maxTransectLength; // end the loop
-				}
-				Intersection.updateIntersectionsWithSubTransect(allIntersections, transect.getOriginPoint(), subTransect, strTree, useFarthest, attGet);
-				if (biasCorrection == null && biasTree != null) {
-					biasCorrection = getBiasValue(subTransect, biasTree, biasGetter);
+			if (planBounds.covers(transect.getOriginPoint())) {
+				Map<DateTime, Intersection> allIntersections = Maps.newHashMap();
```
Comment (isuftin) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java]: What's the advantage of Maps.newHashMap()?

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java [hunk 9]
```diff
@@ -109,70 +141,77 @@ public IntersectionCalculator(SimpleFeatureCollection shorelines, SimpleFeatureC
...
+					transectList.add(feature);
+
+					for (Intersection intersection : allIntersections.values()) {
+						// do I need to worry about order?
+						intersection.setBias(biasCorrection);
+						intersectionList.add(intersection.createFeature(intersectionFeatureType));
+					}
 				}
+			} else {
+				int x = 1;
```
Comment (isuftin) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java]: wat?

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/ShorelineSTRTreeBuilder.java [hunk 10]
```diff
@@ -144,36 +147,37 @@ public ShorelineSTRTreeBuilder(SimpleFeatureCollection shorelines) {
...
-            prevCoord = null;
-        }
-    }
+	/* May also want to take FeatureCollection */
+	private void fillTree(MultiLineString shorelines, SimpleFeature feature) {
+		Coordinate prevCoord = null;
+		for (int i = 0; i < shorelines.getNumGeometries(); i++) {
+			LineString line = (LineString) shorelines.getGeometryN(i);
+			for (Coordinate coord : line.getCoordinates()) {
+				if (prevCoord == null) {
```
Comment (isuftin) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/ShorelineSTRTreeBuilder.java]: Can be rewritten as:

```
...
if (prevCoord != null) {
  LineSegment segment = new LineSegment(prevCoord, coord);
  fillTree(segment, feature, null);
}
prevCoord = coord;
...
```

File: geoserver/dsas-wps/src/test/java/gov/usgs/cida/dsas/hawaii/TestFailingHawaiiTransects.java [hunk 11]
```diff
@@ -74,9 +78,14 @@ public void testExecute() throws Exception {
                 FeatureCollectionFromShp.getFeatureCollectionFromShp(baselineShapefile);
         FeatureCollection<SimpleFeatureType, SimpleFeature> shorelinefc =
                 FeatureCollectionFromShp.getFeatureCollectionFromShp(shorelineShapefile);
-        
+
+        CalculationAreaDescriptor calculationAreaDescriptor = new CalculationAreaDescriptor();
+        calculationAreaDescriptor.setCrs(CRS.decode("EPSG:32604"));
```
Comment (isuftin) [inline geoserver/dsas-wps/src/test/java/gov/usgs/cida/dsas/hawaii/TestFailingHawaiiTransects.java]: Make a descriptive constant for this

**Comment by isuftin** [comment]
Mostly looking good, just added a few comments here and there

File: dsas-web-ui/src/main/webapp/WEB-INF/jsp/stages/transect_verification.jsp (hunk 1 above)
Comment (jiwalker-usgs) [inline dsas-web-ui/src/main/webapp/WEB-INF/jsp/stages/transect_verification.jsp]: no, I didn't change these at all they are still NAD83

File: dsas-web-ui/src/main/webapp/js/stages/transect_verification.js (hunk 2 above)
Comment (jiwalker-usgs) [inline dsas-web-ui/src/main/webapp/js/stages/transect_verification.js]: Yeah, we should work out how to get working UTM zones on map.

File: dsas-web-ui/src/main/webapp/js/stages/transect_verification.js (hunk 3 above)
Comment (jiwalker-usgs) [inline dsas-web-ui/src/main/webapp/js/stages/transect_verification.js]: yup, didn't quite get to checking how errors were handled elsewhere.

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/util/CRSUtils.java (hunk 4 above)
Comment (jiwalker-usgs) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/util/CRSUtils.java]: fair enough

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/util/CRSUtils.java (hunk 5 above)
Comment (jiwalker-usgs) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/util/CRSUtils.java]: nope

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/UpdateTransectsAndIntersectionsProcess.java (hunk 7 above)
Comment (jiwalker-usgs) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/UpdateTransectsAndIntersectionsProcess.java]: yeah, I need to get to this process yet

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java (hunk 8 above)
Comment (jiwalker-usgs) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java]: ???

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java (hunk 9 above)
Comment (jiwalker-usgs) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java]: sorry, this was debugging, didn't realize I checked this in.

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java (hunk 8 above)
Comment (isuftin) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java]: vs new HashMap<>()

File: dsas-web-ui/src/main/webapp/js/stages/transect_verification.js (hunk 2 above)
Comment (isuftin) [inline dsas-web-ui/src/main/webapp/js/stages/transect_verification.js]: Ok. So let's get rid of this. Delete line 56-71, line 74 and line 89

File: geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java (hunk 8 above)
Comment (jiwalker-usgs) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java]: I got the question, don't know the answer, I probably wrote this before we switched to Java 7 and the <> wasn't there yet.

Comment (isuftin) [inline geoserver/dsas-wps/src/main/java/gov/usgs/cida/dsas/wps/geom/IntersectionCalculator.java]: Oh. I thought this was new code.

**Pull request merged**